test/models/hint_test.rb

```diff
@@ -26,6 +35,39 @@ class HintTest < ActiveSupport::TestCase
     assert_nil(match)
   end
 
+  test 'match respects source priority' do
+    ENV['HINT_SOURCES'] = 'custom,aleph'
+
+    match = Hint.match('of science web')
+    assert_equal('custom', match.source)
+
+    hints(:webofscience).destroy
+    match = Hint.match('of science web')
+    assert_equal('aleph', match.source)
+  end
+
+  test 'Hint has default sources if env not set' do
+    ENV.delete('HINT_SOURCES')
```

Comment:
🌈 Sweet. I missed this simple way of removing an set ENV in testing. I have generally resorted to setting the value to none with is way-less-:rainbow: